### PR TITLE
cleanup apiserver wiring using a `With` pattern

### PR DESCRIPTION
**Pull request by deads2k**
@soltysh getting ready for switching.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Why was the openshift-openapi-spec file deleted?

**Comment by deads2k** [comment]
> Why was the openshift-openapi-spec file deleted?

Because we disabled the openapi aggregation.  It finally falls out here https://github.com/openshift/origin/pull/15564#discussion_r130425122 once I separated the API registration.  The swagger fell out because of the gift of a go-restful handler that doesn't properly handle fallthroughs.  It will take some re-plumbing to fix it, but this structure is what we need moving forward.  Hopefully with your changes and disabling during integration tests we can regain it for 3.7 without trying to pick half of 1.8.

**Comment by php-coder** [comment]
Does removal of the `api/swagger-spec/oapi-v1.json` file break `hack/update-generated-swagger-docs.sh` script? If that's the case then how we'll generate API-documentation for our doc site (https://docs.openshift.org/latest/rest_api/openshift_v1.html)?

**Comment by deads2k** [comment]
> Does removal of the api/swagger-spec/oapi-v1.json file break hack/update-generated-swagger-docs.sh script? If that's the case then how we'll generate API-documentation for our doc site (https://docs.openshift.org/latest/rest_api/openshift_v1.html)?

@php-coder who owns that code? Depending on swagger is going to be a problem since, in theory anyway, things are moving to openapi (swagger v2).

**Comment by php-coder** [comment]
>> Does removal of the api/swagger-spec/oapi-v1.json file break hack/update-generated-swagger-docs.sh script?
>>
> @php-coder who owns that code?

I don't know. I see that this file was created by @smarterclayton and modified by @stevekuznetsov

**Comment by stevekuznetsov** [comment]
Unclear what the downstream use of Swagger is ... but if we do not have V1 we need to add V2 before we nuke it

**Comment by soltysh** [comment]
@frobware fyi

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -107,67 +109,114 @@ func (c *MasterConfig) newOpenshiftNonAPIConfig(kubeAPIServerConfig apiserver.Co
...
 
-	openshiftAPIServerConfig, err := c.newOpenshiftAPIConfig(*kubeAPIServerConfig.GenericConfig)
+func (c *MasterConfig) withOpenshiftAPI(delegateAPIServer apiserver.DelegationTarget, kubeAPIServerConfig apiserver.Config) (apiserver.DelegationTarget, error) {
+	openshiftAPIServerConfig, err := c.newOpenshiftAPIConfig(kubeAPIServerConfig)
 	if err != nil {
-		glog.Fatalf("Failed to launch master: %v", err)
+		return nil, err
 	}
-	openshiftAPIServer, err := openshiftAPIServerConfig.Complete().New(openshiftNonAPIServer.GenericAPIServer, stopCh)
+	// We need to add an openshift type to the kube's core storage until at least 3.8.  This does that by using a patch we carry.
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: Can you highlight this with a TODO, maybe.

Comment (deads2k) [inline pkg/cmd/server/origin/master.go]: > Can you highlight this with a TODO, maybe.

It's not a TODO.  We're likely to carry it for all time.

**Comment by stevekuznetsov** [comment]
@deads2k self-lgtm with label add since the bot won't let you do it with a command? shaaaame

**Comment by deads2k** [comment]
> @deads2k self-lgtm with label add since the bot won't let you do it with a command? shaaaame

@stevekuznetsov see the comment right before yours.  I got an lgtm and applied it after a rebase.  Completely reasonable.

**Comment by stevekuznetsov** [comment]
Fair enough :)

**Pull request merged**